lua/wire/wire_paths.lua

```diff
@@ -0,0 +1,116 @@
+-- wire_paths.lua
+--
+-- This file implements syncing of wire paths, which are the visual
+-- component of wires.
+--
+-- Conceptually, a wire path has a material, a color, and a non-zero width, as
+-- well as as a non-empty polyline along the wire. (Each point in the line
+-- has both a parent entity, and a local offset from that entity.)
+--
+
+if not WireLib then return end
+WireLib.Paths = {}
+
+local transmit_queues = setmetatable({}, { __index = function(t,p) t[p] = {} return t[p] end })
+
+if CLIENT then
+	net.Receive("WireLib.Paths.TransmitPath", function(length)
+		local path = {
+			Path = {}
+		}
+		path.Entity = net.ReadEntity()
+		path.Name = net.ReadString()
+		path.Width = net.ReadFloat()
+		if path.Width<=0 then
+			if path.Entity.WirePaths then
+				path.Entity.WirePaths[path.Name] = nil
+				if not next(path.Entity.WirePaths) then path.Entity.WirePaths = nil end
+			end
+			return
+		end
+		path.StartPos = net.ReadVector()
+		path.Material = net.ReadString()
+		path.Color = net.ReadColor()
+
+		local num_points = net.ReadUInt(15)
+		for i = 1, num_points do
+			path.Path[i] = { Entity = net.ReadEntity(), Pos = net.ReadVector() }
+		end
+
+		if path.Entity.WirePaths == nil then path.Entity.WirePaths = {} end
+		path.Entity.WirePaths[path.Name] = path
```

Comment:
I see. Looks like that callback needs IsValid check too. I'll add it once I get home. Would Entity:IsValid() return false in the outside pvs case too?